[CI] CrossClusterQueryUnavailableRemotesIT testRemoteOnlyCCSAgainstDisconnectedRemoteWithSkipUnavailableTrue failing

**Build Scans:**
- [elasticsearch-intake #17133 / part3](https://gradle-enterprise.elastic.co/s/gjy6edgrfrxj6)
- [elasticsearch-pull-request #54735 / part-3](https://gradle-enterprise.elastic.co/s/lof52c42kbpeg)
- [elasticsearch-pull-request #54732 / part-3](https://gradle-enterprise.elastic.co/s/ov32vpbnsb54i)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.action.CrossClusterQueryUnavailableRemotesIT.testRemoteOnlyCCSAgainstDisconnectedRemoteWithSkipUnavailableTrue" -Dtests.seed=CD8B73F6EBD82287 -Dtests.locale=chr-Cher-US -Dtests.timezone=Europe/Sofia -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.action.CrossClusterQueryUnavailableRemotesIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testRemoteOnlyCCSAgainstDisconnectedRemoteWithSkipUnavailableTrue),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <3>
     but: was <1>
```

**Issue Reasons:**
- [main] 3 consecutive failures in test testRemoteOnlyCCSAgainstDisconnectedRemoteWithSkipUnavailableTrue
- [main] 2 consecutive failures in step part-3
- [main] 2 consecutive failures in pipeline elasticsearch-pull-request
- [main] 3 failures in test testRemoteOnlyCCSAgainstDisconnectedRemoteWithSkipUnavailableTrue (1.7% fail rate in 178 executions)
- [main] 2 failures in step part-3 (2.7% fail rate in 75 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (2.6% fail rate in 77 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -367,9 +367,6 @@ tests:
 - class: org.elasticsearch.xpack.ml.integration.ClassificationIT
   method: testDependentVariableIsAliasToKeyword
   issue: https://github.com/elastic/elasticsearch/issues/121492
-- class: org.elasticsearch.xpack.esql.action.CrossClusterQueryUnavailableRemotesIT
-  method: testRemoteOnlyCCSAgainstDisconnectedRemoteWithSkipUnavailableTrue
-  issue: https://github.com/elastic/elasticsearch/issues/121578
 - class: org.elasticsearch.xpack.esql.action.CrossClustersCancellationIT
   method: testTasks
   issue: https://github.com/elastic/elasticsearch/issues/121626

